src/Tasks.h

```diff
@@ -446,6 +510,94 @@ class MFNetflixTask : public MLTask {
    std::unique_ptr<PSSparseServerInterface> psint;
 };
 
+class LDATaskS3 : public MLTask {
+ public:
+  LDATaskS3(uint64_t model_size,
+            uint64_t batch_size,
+            uint64_t samples_per_batch,
+            uint64_t features_per_sample,
+            uint64_t nworkers,
+            uint64_t worker_id,
+            const std::string& ps_ip,
+            uint64_t ps_port)
+      : MLTask(model_size,
+               batch_size,
+               samples_per_batch,
+               features_per_sample,
+               nworkers,
+               worker_id,
+               ps_ip,
+               ps_port),
+        psint(nullptr) {}
+  /**
+   * Worker here is a value 0..nworkers - 1
+   */
+  void run(const Configuration& config, int worker, int test_iters);
+
+ private:
+  /**
+   * Helper function to push the doc-topic statistics to S3
+   */
+  void upload_wih_bucket_id_fn(char* mem_to_send,
+                               uint64_t to_send_size,
+                               int& upload_lock,
+                               int bucket_id);
+  /**
+   * Helper function to push the update to S3
+   */
+  void push_gradient(char* gradient_mem,
+                     int total_sampled_tokens,
+                     int total_sampled_docs,
+                     uint32_t to_send_size);
+  /**
+   * Load the pre-cached token indices (for the current slice)
+   * from the server
+   */
+  void load_serialized_indices(char* mem_begin);
+
+  std::vector<std::unique_ptr<std::thread>> help_upload_threads;
```

Comment:
Doesn't seem used

Reply:
It's being used now. Previously `help_upload_threads` was removed since in all the experiments I ensured that each worker was assigned with exactly one LDAStatistics and thus it was not needed.

Reply:
Work has been done to ensure it works fine.